### PR TITLE
Remove cookie-jar dependency for ruby 3.3

### LINKED ISSUE #2394: Ruby 3.3 && cookiejar = ArgumentError
While testing Ruby 3.3, `cookiejar` doesn't seem to work. `rspec` fails with
```
ArgumentError:
  wrong number of arguments (given 3, expected 1..2)
# /usr/local/bundle/gems/ruby/3.3.0/gems/cookiejar-0.3.3/lib/cookiejar/cookie_validation.rb:48:in `initialize'
# /usr/local/bundle/gems/ruby/3.3.0/gems/cookiejar-0.3.3/lib/cookiejar/cookie_validation.rb:48:in `new'
# /usr/local/bundle/gems/ruby/3.3.0/gems/cookiejar-0.3.3/lib/cookiejar/cookie_validation.rb:48:in `<module:CookieValidation>'
# /usr/local/bundle/gems/ruby/3.3.0/gems/cookiejar-0.3.3/lib/cookiejar/cookie_validation.rb:25:in `<module:CookieJar>'
# /usr/local/bundle/gems/ruby/3.3.0/gems/cookiejar-0.3.3/lib/cookiejar/cookie_validation.rb:5:in `<top (required)>'
# /usr/local/bundle/gems/ruby/3.3.0/gems/cookiejar-0.3.3/lib/cookiejar/cookie.rb:3:in `<top (required)>'
# /usr/local/bundle/gems/ruby/3.3.0/gems/cookiejar-0.3.3/lib/cookiejar.rb:1:in `<top (required)>'
# /usr/local/bundle/gems/bundler-2.4.22/lib/bundler.rb:187:in `require'
# ./spec/spec_helper.rb:9:in `<top (required)>'
```

It's only used in test and we can easily isolate the specs.

### PR DESCRIPTION
**Pull request by ericproulx**
Fix #2394

This PR also changes our `cookie[...].delete` data. I followed [Rack::Utils.delete_set_cookie_header](https://github.com/rack/rack/blob/64ad26e3381da2ce1853638a2c4ea241c2ad3729/lib/rack/utils.rb#L338-L340). At some point, we should refactor our `Grape::Cookies` to use the function instead of explicitly setting the values.

Documentation about [Max-Age](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#max-agenumber)

### CODE REVIEW

**Comment by ericproulx** [comment]
I haven't found a good candidate to replace `cookiejar`, so I've created a helper to manage common cases and different version of `rack`. Speaking of `rack`, we might consider dropping `rack < 2` since we dropped Rails 5.2 and Ruby 2.6. Starting from Rails 6, it's `rack >=2`.

**Comment by ericproulx** [comment]
@dblock something is going on with `danger`

> bundler: failed to load command: danger (/home/runner/work/grape/grape/vendor/bundle/ruby/2.7.0/bin/danger)
> /home/runner/work/grape/grape/vendor/bundle/ruby/2.[7](https://github.com/ruby-grape/grape/actions/runs/7357537616/job/20029311581?pr=2395#step:4:8).0/gems/octokit-4.25.1/lib/octokit/response/raise_error.rb:14:in `on_complete': GET https://api.github.com/repos/ruby-grape/grape/pulls/2395: 403 - API rate limit exceeded for user ID 20649811. If you reach out to GitHub Support for help, please include the request ID 2848:17B1:4FF5E1:A82AE0:658ED9FE. // See: https://docs.github.com/rest/overview/rate-limits-for-the-rest-api (Octokit::TooManyRequests)

File: spec/support/cookie_helper.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+module Spec
+  module Support
+    module Helpers
+      def cookie_expires_value
+        @cookie_expires_value ||= Gem::Version.new(Rack.release) <= Gem::Version.new('2') ? Time.at(0).gmtime.rfc2822 : Time.at(0).httpdate
+      end
+
+      def last_response_cookies
+        Array(last_response.headers['Set-Cookie']).flat_map { |h| h.split("\n") }.sort
+      end
```
Comment (dblock) [inline spec/support/cookie_helper.rb:12]: If you want to make this more elegant, you could monkey patch `last_response` to get `last_response.cookies` and `last_response.cookies.expires_at` or something like that.

Comment (ericproulx) [reply spec/support/cookie_helper.rb:12]: Its already defined with Rack 3.0 and its using CGI:Cookie to parse cookies. Unfortunately, it doesn't manage `max-age`

Comment (dblock) [reply spec/support/cookie_helper.rb:12]: Exactly. When using an older version of Rack, define the same method as Rack 3.0 by including a monkey patch in https://github.com/ruby-grape/grape/tree/master/spec/support.

https://github.com/rack/rack/blob/64ad26e3381da2ce1853638a2c4ea241c2ad3729/lib/rack/request.rb#L293

**Review by dblock — APPROVED** [review]
This looks nice and clean and extensible! Do update CHANGELOG even if it's test work.

**Comment by ericproulx** [comment]
> This looks nice and clean and extensible! Do update CHANGELOG even if it's test work.

Done

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
 * [#2384](https://github.com/ruby-grape/grape/pull/2384): Allow to use `before/after/rescue_from` methods in any order when using `mount` - [@jcagarcia](https://github.com/jcagarcia).
 * [#2390](https://github.com/ruby-grape/grape/pull/2390): Drop support for Ruby 2.6 and Rails 5 - [@ericproulx](https://github.com/ericproulx).
 * [#2393](https://github.com/ruby-grape/grape/pull/2393): Optimize AttributeTranslator - [@ericproulx](https://github.com/ericproulx).
+* [#2395](https://github.com/ruby-grape/grape/pull/2395): Remove cookiejar dependency for ruby 3.3 - [@ericproulx](https://github.com/ericproulx).
```
Comment (dblock) [inline CHANGELOG.md:12]: Re-reading the code, I think we hid a change here, cookiejar isn't the important one. 

Let's add another line, maybe "set `max_age=0` in `cookies.delete`"?

**Pull request merged**